### ISSUE TITLE
Exec works only with Promise

### ISSUE THREAD

**Issue by AeonRush** [whitecolor/mongoose-fill#13]
Hi
When I tried to write this:
```javascript
Art.find().fill('likes').exec(function (err, data) {
  res.send(data);
});
```
I didn't got response. But with this:
```javascript
Art.find().fill('likesCount').exec().then(function (data) {
  res.send(data);
});
```
all works fine.
And, can you explain how to catch errors with this? Will this work?
```javascript
Art.find().fill('likesCount').exec().then(cbOk,  cbError);
```
Thanks

**Comment by whitecolor** [comment]
Hi, I'm not sure why this:
```js
Art.find().fill('likesCount').exec().then(function (data) {
  res.send(data);
});
```
doesn't work, `mongoose-fill` seem not to mess with that mechanics.

> And, can you explain how to catch errors with this? Will this work?

Yes will work, this is usual way to deal with errors when using promises.

**Comment by whitecolor** [comment]
Ah yes you probably right, I will look into it.

**Comment by whitecolor** [comment]
Fixed in 1.6
